examples/example.hpp

```diff
@@ -1070,3 +1070,21 @@ void register_glfw_callbacks(window& app, glfw_state& app_state)
         }
     };
 }
+
+// Find devices with specified streams
+std::string depth_with_stream_type_present(rs2_stream type)
```

Comment:
Refactor into `bool device_with_streams(std::vector<RS2_STREAM_TYPE>, std::string& out_serial)` signature to cover all the required cases